### PR TITLE
feat: typescript config

### PR DESCRIPTION
**Pull request by TheBorsuk**
Added TypeScript eslint config file.

### CODE REVIEW

File: frontend/typescript/index.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+module.exports = {
+    root: true,
+    extends: [
+        'airbnb-typescript',
```
Comment (rafalswietek) [inline frontend/typescript/index.js:4]: Shouldn't we extend our regular js front-end config, or at least copy its rules? 🤔

Comment (TheBorsuk) [reply frontend/typescript/index.js:4]: Yup, it was there, don't know why it was deleted 🤦

File: frontend/typescript/index.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        ecmaVersion: 2020,
+        project: 'tsconfig.json',
+    },
+    overrides: [
+        {
+            files: ["**/*.{ts,tsx}"],
+            rules: {
+                "complexity": ["error", 10],
+                "@typescript-eslint/explicit-function-return-type": ['error'],
+                "@typescript-eslint/no-explicit-any": ['error'],
```
Comment (aliaksandr-savitski) [inline frontend/typescript/index.js:19]: **RC:** Please separate `complexity` rule to match only`tsx`. We override complexity rule only for JSX (tsx) files.

**Review by aliaksandr-savitski — CHANGES_REQUESTED** [review]
Just one small thing from my side

**Review by aliaksandr-savitski — APPROVED** [review]
🍺

File: package.json [hunk 3]
```diff
@@ -14,19 +14,20 @@
...
-    "eslint-plugin-import": "^2.22.0",
-    "eslint-plugin-jsx-a11y": "^6.3.1",
-    "eslint-plugin-react": "^7.20.6",
-    "eslint-plugin-react-hooks": "^4.1.0"
+    "eslint-config-airbnb-typescript": "^12.0.0",
+    "eslint-plugin-import": "^2.22.1",
+    "eslint-plugin-jsx-a11y": "^6.4.1",
+    "eslint-plugin-react": "^7.21.5",
+    "eslint-plugin-react-hooks": "^4.2.0",
+    "typescript": "^4.1.2"
```
Comment (Necromos) [inline package.json:30]: **Q** Do we need typescript as peer here? 🤔

File: package.json [hunk 4]
```diff
@@ -14,19 +14,20 @@
...
-  "devDependencies": {
-    "eslint": "^7.7.0"
-  },
   "peerDependencies": {
+    "@typescript-eslint/eslint-plugin": "^4.8.1",
+    "@typescript-eslint/parser": "^4.8.1",
     "@babel/core": "^7.11.1",
     "@babel/eslint-parser": "^7.11.3",
-    "eslint": "^7.7.0",
+    "eslint": "^7.14.0",
```
Comment (Necromos) [inline package.json:22]: **Q** Is it required? I would be opposed to bumping this up if that's not crucial due to the fact of current usage 🤔

File: package.json [hunk 5]
```diff
@@ -14,19 +14,20 @@
...
     "eslint-config-airbnb-base": "^14.2.0",
-    "eslint-plugin-import": "^2.22.0",
-    "eslint-plugin-jsx-a11y": "^6.3.1",
-    "eslint-plugin-react": "^7.20.6",
-    "eslint-plugin-react-hooks": "^4.1.0"
+    "eslint-config-airbnb-typescript": "^12.0.0",
+    "eslint-plugin-import": "^2.22.1",
+    "eslint-plugin-jsx-a11y": "^6.4.1",
+    "eslint-plugin-react": "^7.21.5",
+    "eslint-plugin-react-hooks": "^4.2.0",
```
Comment (Necromos) [inline package.json:29]: **Q** Is it required? I would be opposed to bumping this up if that's not crucial due to the fact of current usage 🤔

File: package.json [hunk 6]
```diff
@@ -14,19 +14,20 @@
   "author": "MasterBorn Sp. z o.o.",
   "license": "MIT",
   "dependencies": {},
-  "devDependencies": {
-    "eslint": "^7.7.0"
-  },
```
Comment (Necromos) [inline package.json:19]: **Q** Why that? 🤔

File: tsconfig.template.json [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    "noEmit": true
+  },
+  "exclude": [
+    "node_modules",
+    "build"
+  ],
+  "include": [
+    "src"
+  ]
+}
```
Comment (Necromos) [inline tsconfig.template.json:39]: **RC(C)** Can we store this one somewhere else? :) This repo should be used for eslint only, so adding TS config here will make chaos actually :) For the time being, you can post this one on Confluence within standards for general discussion on it and then we can store it as some template or sth :)

**Review by Necromos — CHANGES_REQUESTED** [review]
Few questions + concern about tsconfig placement :)

**Review by Necromos — APPROVED** [review]
👍

**Pull request merged**